lib/active_merchant/billing/gateways/mundipagg.rb

```diff
@@ -288,14 +304,26 @@ def commit(action, parameters, auth = nil)
         )
       end
 
-      def success_from(response)
-        %w[pending paid processing canceled active].include? response['status']
+      def extract_charge_object_from_response(response, url)
+        orders_endpoint?(url) ? charge_object(response) : response
+      end
+
+      def charge_object(response)
+        response.try(:[], 'charges').try(:[], 0)
```

Comment:
I would consider `response.dig('charges', 0)` here. It should effectively be the same as what you have, but a bit more readable.

Reply:
Yes, definitely an improvement. Thank you, I will change it.